consensus/ethash/consensus.go

```diff
@@ -673,7 +674,8 @@ func accumulateRewards(config *params.ChainConfig, state *state.StateDB, header
 	}
 	if config.IsCancun(header.Number) {
 		coinbase := utils.GetTreeKeyBalance(header.Coinbase.Bytes())
-		state.Witness().TouchAddress(coinbase, state.GetBalance(header.Coinbase).Bytes())
+		state.Witness().TouchAddressOnReadAndChargeGas(coinbase)
```

Comment:
It looks weird but the gas returned from `TouchAddressOnReadAndChargeGas` is not subtracted so it's fine.  Everywhere else I follow the pattern of `TouchAddress*AndChargeGas` to set the key and then `SetLeafValue` to set the value.  If I just call `SetLeafValue` here it will panic (https://github.com/gballet/go-ethereum/pull/63/files#diff-76413e8d8e3b91ef8ab764d915b3cd96c12dea41a16a67952cd1198fc194c744R73).

Having this check panic is probably a bit severe.  But it also serves as a check to ensure that the code is correctly implemented (every key/value added to the `AccessWitness` should be charged for, so a correct implementation will never reach the panic).